test/es-module/test-http-imports-cli.mjs

```diff
@@ -1,48 +1,48 @@
-import { mustCall } from '../common/index.mjs';
-import { match, notStrictEqual } from 'assert';
-import { spawn } from 'child_process';
-import { execPath } from 'process';
-
-{
-  const child = spawn(execPath, [
-    '--experimental-network-imports',
-    '--input-type=module',
-    '-e',
-    'import "http://example.com"',
-  ]);
-
-  let stderr = '';
-  child.stderr.setEncoding('utf8');
-  child.stderr.on('data', (data) => {
-    stderr += data;
-  });
-  child.on('close', mustCall((code, _signal) => {
+import { mustCall, spawnPromisified } from '../common/index.mjs';
+import { match, notStrictEqual } from 'node:assert';
+import { spawn as spawnAsync } from 'node:child_process';
+import { execPath } from 'node:process';
+import { describe, it } from 'node:test';
+
+
+describe('ESM: http import via CLI', { concurrency: true }, () => {
+  const disallowedSpecifier = 'http://example.com';
+
+  it('should throw disallowed error for insecure protocol', async () => {
+    const { code, stderr } = await spawnPromisified(execPath, [
+      '--experimental-network-imports',
+      '--input-type=module',
+      '--eval',
+      `import "${disallowedSpecifier}"`,
+    ]);
+
     notStrictEqual(code, 0);
 
     // [ERR_NETWORK_IMPORT_DISALLOWED]: import of 'http://example.com/' by
     //   …/[eval1] is not supported: http can only be used to load local
     // resources (use https instead).
     match(stderr, /[ERR_NETWORK_IMPORT_DISALLOWED]/);
-  }));
-}
-{
-  const child = spawn(execPath, [
-    '--experimental-network-imports',
-    '--input-type=module',
-  ]);
-  child.stdin.end('import "http://example.com"');
-
-  let stderr = '';
-  child.stderr.setEncoding('utf8');
-  child.stderr.on('data', (data) => {
-    stderr += data;
+    match(stderr, new RegExp(disallowedSpecifier));
   });
-  child.on('close', mustCall((code, _signal) => {
-    notStrictEqual(code, 0);
 
-    // [ERR_NETWORK_IMPORT_DISALLOWED]: import of 'http://example.com/' by
-    //   …/[stdin] is not supported: http can only be used to load local
-    // resources (use https instead).
-    match(stderr, /[ERR_NETWORK_IMPORT_DISALLOWED]/);
-  }));
-}
+  it('should throw disallowed error for insecure protocol in REPL', () => {
+    const child = spawnAsync(execPath, [
+      '--experimental-network-imports',
+      '--input-type=module',
+    ]);
+    child.stdin.end(`import "${disallowedSpecifier}"`);
+
+    let stderr = '';
+    child.stderr.setEncoding('utf8');
+    child.stderr.on('data', (data) => stderr += data);
+    child.on('close', mustCall((code, _signal) => {
+      notStrictEqual(code, 0);
+
+      // [ERR_NETWORK_IMPORT_DISALLOWED]: import of 'http://example.com/' by
+      //   …/[stdin] is not supported: http can only be used to load local
+      // resources (use https instead).
+      match(stderr, /[ERR_NETWORK_IMPORT_DISALLOWED]/);
+      match(stderr, new RegExp(disallowedSpecifier));
```

Comment:
That seems quite unsafe, the dot is interpreted as "any character" instead of an actual dot. If it's important that the error message contains the specifier, let's make an actual regex, or use `stderr.includes(disallowedSpecifier)`.

Reply:
`[` and `]` are special characters in regex, I think this would need to be `/\[ERR_NETWORK_IMPORT_DISALLOWED\]/`

Reply:
I considered declaring `disallowedSpecifier` as a regex, but then that makes consuming it in the import statement hairy. I don't like `assert.ok()` for things like this because the thrown AssertionError is pretty unhelpful, but I think it's better than the alternative :/ Has there been any discussion about adding an `assert.includes()`?

> `[` and `]` are special characters in regex

🤦‍♂️ Thank you

Reply:
> Has there been any discussion about adding an `assert.includes()`?

Sounds like the kind of things that did not happen so far because no one did it.